w03h03/test/pgdp/MoreSeamCarvingTests.java

```diff
@@ -0,0 +1,467 @@
+            int[] image1 = {   //the format of the hex codes is mixed on purpose to make sure every valid entry (w and w/o transparency) works
+                           0x0, 0x12bbff37,   0x2800ed,
+                    0xffff0064,   0xff7f24, 0xffffffff,
+                      0x49b8ff, 0x46ff0064, 0x00ff2424,
+            };
+            int width = 3;
+            int height = 3;
+            int[] gradientMagnitudesOfFlatImage = new int[width*height];
+            var seamcarving = new SeamCarving();
+            seamcarving.toGradientMagnitude(image1, gradientMagnitudesOfFlatImage, width, height);
+
+            int[] image1_result = new int[] {
+                    2147483647, 2147483647, 2147483647,
+                    2147483647,     160724, 2147483647,
+                    2147483647, 2147483647, 2147483647
+            };
+            assertArrayEquals(image1_result, gradientMagnitudesOfFlatImage);
+        }
+
+        @Test
+    /*
+        TODO check manually whether the result is actually correct; this currently just tests whether image2 gives the
+         output expected assuming that the method is correct in its current state
+         (where it correctly works with toGradientMagnitude3x3Image), basically to see whether changing the method
+         changes the output
+     */
+        void toGradientMagnitude5x6Image(){
+            int[] image2 = {   //the format of the hex codes is mixed on purpose to make sure every valid entry (w and w/o transparency) works
+                      0xffffff,   0x780000,      0x000,        0x0,  0xfffffff,
+                    0xffffffff,        0x0, 0x23ff1291,   0x008200, 0xffffe6c8,
+                      0xffffff,   0xdbdeff,  0xf6dff9f,   0xffffff,   0xffffff,
+                      0x6f12ff, 0x22ff9424,   0xbfc800,   0xda80ff,        0x0,
+                    0xffffffff, 0xdeffffff,       0x00,   0xffffff,  0x56912ff,
+                    0x01ffffff,        0x0,   0xffffff,   0x80caff,  0xfffffff
+            };
+            int width = 5;
+            int height = 6;
+            int[] gradientMagnitudesOfFlatImage = new int[width*height];
+            var seamcarving = new SeamCarving();
+            seamcarving.toGradientMagnitude(image2, gradientMagnitudesOfFlatImage, width, height);
+
+            int[] image2_result = new int[] {
+                    2147483647, 2147483647, 2147483647, 2147483647, 2147483647,
+                    2147483647,     192379,     119087,     243044, 2147483647,
+                    2147483647,     118757,      60630,     143085, 2147483647,
+                    2147483647,     106934,     151917,      76481, 2147483647,
+                    2147483647,     283300,      72146,      89950, 2147483647,
+                    2147483647, 2147483647, 2147483647, 2147483647, 2147483647
+            };
+            assertArrayEquals(image2_result, gradientMagnitudesOfFlatImage);
+        }
+
+        @Test
+        void toGradientMagnitude0x6Image(){
```

Comment:
I'm not really convinced of the usefullness of your 0x6 test...

Your test does not need an assertion because `toGradientMagnitude()` can't alter the contents of an array with `length == 0`. Youre only testing if the method call causes an `ArrayIndexOutofBoundsexception`.

I think we should test if `shrink(new int[?], new int[?], 0, ?, ?) -> new int[0]` and `shrink(new int[?], new int[?], ?, 0, ?) -> new int[0]`. This would cover all relevant test cases. (my implementation doesn't pass your test because I assume every call to `toGradientMagnitude()` to be valid)

Reply:
+1